clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -749,58 +869,104 @@ UnsentRequest buildRequest(long currentTimeMs) {
 
             BiConsumer<ClientResponse, Throwable> responseHandler = (clientResponse, error) -> {
                 if (error != null) {
-                    onFailedAttempt(currentTimeMs);
                     errorHandler.handle(nodeToSend, requestBuilder.data(), this, error, currentTimeMs);
                     processingComplete();
                 } else {
                     successHandler.handle(nodeToSend, requestBuilder.data(), this, clientResponse, currentTimeMs);
+                    if (onClose && !closeFuture.isDone()) {
+                        closeFuture.complete(null);
+                    }
                 }
             };
 
             if (requestBuilder == null) {
-                log.trace("Building ShareAcknowledge request to send to node {} failed", nodeToSend.id());
                 handleSessionErrorCode(Errors.SHARE_SESSION_NOT_FOUND);
                 return null;
             } else {
-                log.trace("Building ShareAcknowledge request to send to node {}", nodeToSend.id());
+                inFlightAcknowledgements.putAll(finalAcknowledgementsToSend);
+                if (incompleteAcknowledgements.isEmpty()) {
+                    acknowledgementsToSend.clear();
+                } else {
+                    incompleteAcknowledgements.clear();
```

Comment:
This line may not be required.

Reply:
Yes, it was not required, I have removed it now.